tests/test-maximize.lua

```diff
@@ -0,0 +1,91 @@
+--- Tests maximize and fullscreen
+-- Test for https://github.com/awesomeWM/awesome/issues/134.
+
+local runner = require("_runner")
+local awful = require("awful")
+
+local original_geo = nil
+
+local steps = {
+    function(count)
+        if count == 1 then
+            awful.spawn("xterm")
+        else
+            local c = client.get()[1]
+            if c then
+                original_geo = c:geometry()
+                return true
+            end
+        end
+    end,
+
+    -- maximize horizontally
+    function()
+        local c = client.get()[1]
+        assert(not c.maximized_horizontal)
+        assert(not c.maximized_vertical  )
+        assert(not c.fullscreen          )
+
+        c.maximized_horizontal = true
+        return true
+    end,
+    function()
+        local c = client.get()[1]
+
+        local new_geo = c:geometry()
+        local sgeo = c.screen.workarea
+
+        --assert(new_geo.x-c.border_width==sgeo.x) --FIXME c:geometry({x=1}).x ~= 1
+
+        --assert(new_geo.width+2*c.border_width==sgeo.width) --FIXME off by 4px
```

Comment:
I think these asserts should work before this is merged. Otherwise, you could just skip&remove this part of the test.